OpenRA.Mods.Common/Traits/SupportPowers/AirstrikePower.cs

```diff
@@ -105,23 +101,18 @@ public void SendAirstrike(Actor self, WPos target, bool randomize = true, int at
 
 				// Remove the camera when the final plane leaves the target area
 				if (!aircraftInRange.Any(kv => kv.Value))
-				{
-					if (camera != null)
-					{
-						camera.QueueActivity(new Wait(info.CameraRemoveDelay));
-						camera.QueueActivity(new RemoveSelf());
-					}
-
-					camera = null;
+					RemoveCamera(camera);
+			};
 
-					if (beacon != null)
-					{
-						self.World.AddFrameEndTask(w =>
-						{
-							w.Remove(beacon);
-							beacon = null;
-						});
-					}
+			Action<Actor> onRemovedFromWorld = a =>
+			{
+				// Checking for attack range is not relevant here because
+				// aircraft may be shot down before entering. Thus we remove
+				// the camera and beacon only if the whole squad is dead.
+				if (!aircraftInRange.Any(kv => !kv.Key.IsDead))
```

Comment:
Minor style nit: I'd find `if (aircraftInRange.All(kv => kv.Key.IsDead))` easier to mentally parse than that double negation.